src/platform.hpp

```diff
@@ -0,0 +1,16 @@
+// Copyright © 2024 Dylan Baker
+// SPDX-License-Identifier: MIT
+
+/**
+ * Helpers to abstract the target platform.
+*/
+
+#pragma once
+
+#include <filesystem>
+
+namespace platform {
+
+    const std::filesystem::path libdir();
```

Comment:
We can drop the const qualifier on this. It's return by copy anyway. Corrupting the original libdir value isn't possible.